### PR TITLE
feat(rc): provide ui related custom configuration

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

cherry-pick of https://github.com/wireapp/wire-android-reloaded/pull/1664

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by alexandreferris — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/config/CustomUiConfigurationProvider.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import androidx.compose.runtime.staticCompositionLocalOf
+import com.wire.android.BuildConfig
+
+/**
+ * A provider which provides flags for the custom build configuration related to the UI
+ *
+ */
+object CustomUiConfigurationProvider {
+    val isAccountCreationAllowed = BuildConfig.ALLOW_ACCOUNT_CREATION ?: true
+}
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/config/CustomUiConfigurationProvider.kt:29]: How does it differ from the `FeatureVisibilityFlags` that we already have? Couldn't we just add this `isAccountCreationAllowed` there? 🤔

Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/config/CustomUiConfigurationProvider.kt:29]: It differs in a sense that this is coming from the config and that other one is just something we will ditch later in the future

Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/config/CustomUiConfigurationProvider.kt:29]: I was unaware of this other `FeatureVisibilityFlags`. But it does seem to be a "temporary" thing, as features are being developed. In any case, I guess it's indeed different from this one and can be left aside.

The overview is that:

Some flags might come from build time (the `default.json` or `custom.json` file).
Some might come from the backend (either backend settings or team settings).

Going forward, we should have a single place to handle UI flags and it shouldn't matter where they come from. Use the current user + Kalium's user scope + build flags and centralise everything into a single flag provider.

**Review by vitorhugods — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/config/CustomUiConfigurationProvider.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/config/CustomUiConfigurationProvider.kt:29]: We could just not treat it as a temporary solution, for me it's the same, it shouldn't matter where the flags come from and actually `FeatureVisibilityFlags` fits better this purpose, as we're enabling or disabling a feature of creating account 😄 
But ok, it's not that much important, just for me at least, having two very similar objects that are basically responsible for the same thing makes it a little bit more confusing 😉

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1699/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.0(6312063)(6312063)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.0(6312063)(6312063)-staging-compat.apk)

**Pull request merged**